src/dmd/root/longdouble.d

```diff
@@ -6,29 +6,30 @@
  * https://github.com/dlang/dmd/blob/master/src/root/longdouble.d
  */
 
-// 80 bit floating point value implementation for LDC compiler targetting MSVC
+// 80-bit floating point value implementation if the C/D compiler do not support them natively
 
 module dmd.root.longdouble;
 
-enum hasNativeFloat80 = real.sizeof > 8;
-
-static if (hasNativeFloat80)
-{
+static if (real.sizeof > 8)
     alias longdouble = real;
-}
+else
+    alias longdouble = longdouble_soft;
 
-static if (!hasNativeFloat80):
+// longdouble_soft needed when building the backend with
+// Visual C or the frontend with LDC on Windows
+version(CRuntime_Microsoft):
 extern(C++):
 nothrow:
+@nogc:
```

Comment:
Investigate using `pure` and `@safe` as much as practical.

Reply:
Added `pure` and `@safe`, but it has `pure @trusted` has to be verified manually on all the asm.